lab-sharmarke/route/car-router.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const debug = require('debug')('car:car-router');
+const Car = require('../model/car.js');
+const carRouter = new Router();
+
+carRouter.post('/api/note', jsonParser, function(req, res, next) {
+  debug('POST: /api/car');
+
+  Car.createCar(req.body)
+  .then( car => res.json(car))
+  .catch( err => next(err));
+});
+
+carRouter.get('/api/car/:id', function(req, res, next) {
+  debug('GET: /api/car/:id');
+
+  Car.fetchCar(req.params.id)
+  .then( car => res.json(car))
+  .catch( err => next(err));
+});
+
+carRouter.get('/api/car', function(req, res, next) {
+  debug('GET: /api/car');
+
+  Car.fetchIDs()
+  .then( ids => res.json(ids))
+  .catch( err => next(err));
+});
+
+carRouter.put('/api/car', jsonParser, function(req, res, next) {
+  debug('POST: /api/car');
+
+  Car.updateCar(req.quer.id, req.body)
```

Comment:
this route you probably want to define the id as a url param as opposed to a query.